R/fiveyeargwsum-render.R

```diff
@@ -1,100 +1,151 @@
+
+  #Create the Base Plot Object
+  plot_object <- gsplot(yaxs = 'i', ylog=logAxis, xaxt = "n", mar = c(8, 4, 4, 2) + 0.1) %>%
+      axis(side = 1, at = date_seq_mo, labels = FALSE) %>%
+      view(xlim = c(startDate, endDate)) %>%
       legend(location = "below", cex = 0.8, ncol = 2, y.intersp = 1.5) %>%
-      axis(side = 2, reverse = isInverted) %>%
+      axis(side = 2, reverse = invertedFlag) %>%
       grid(col = "lightgrey", lty = 1) %>%
       title(ylab = "Water Level, Below LSD (feet)")
 
-    plot_object <-
-      XAxisLabels(plot_object,
-                  fiveyrInfo$month_label,
-                  fiveyrInfo$month_label_location,
-                  fiveyrInfo$date_seq_yr + months(6))
-    
-    for (i in grep("^appr_.*_uv$", names(fiveyrData), invert = TRUE)) {
-      fiveyrPlotConfig <-
-        getFiveyearPlotConfig(fiveyrData[i], fiveyrInfo, maxLabel = maxLabel, minLabel = minLabel)
-      for (j in seq_len(length(fiveyrPlotConfig))) {
-        plot_object <-
-          do.call(names(fiveyrPlotConfig[j]), append(list(object = plot_object), fiveyrPlotConfig[[j]]))
-      }
-    }
-    
-    # add vertical lines to delineate calendar year boundaries
-    plot_object <- DelineateYearBoundaries(plot_object, fiveyrInfo$date_seq_yr)
-    
-    plot_object <- ApplyApprovalBarStyles(plot_object, fiveyrData)
+  plot_object <- 
+    XAxisLabels(plot_object, month_label, month_label_location, date_seq_yr + months(6))
+
+  #Plot Time Series
+  plot_object <- plotTimeSeries(plot_object, statTimeSeries, 'statTimeSeries', timezone, getFiveYearPlotConfig, isDV=TRUE)
+  plot_object <- plotTimeSeries(plot_object, statTimeSeriesEst, 'statTimeSeriesEst', timezone, getFiveYearPlotConfig, isDV=TRUE)
+
+  #Plot Other Items
+  plot_object <- plotItem(plot_object, groundWaterLevels, 'gw_level', getFiveYearPlotConfig, isDV=TRUE)
+  plot_object <- plotItem(plot_object, minMaxPoints[['min_iv']], 'min_iv', getFiveYearPlotConfig, isDV=TRUE)
+  plot_object <- plotItem(plot_object, minMaxPoints[['max_iv']], 'max_iv', getFiveYearPlotConfig, isDV=TRUE)
+
+  # add vertical lines to delineate calendar year boundaries
+  plot_object <- DelineateYearBoundaries(plot_object, date_seq_yr)
+  
+  plot_object <- ApplyApprovalBarStyles(plot_object, approvals)
+  
+  plot_object <- rmDuplicateLegendItems(plot_object)
+  
+  # Add space to the top of the Y Axis
+  plot_object <- RescaleYTop(plot_object)
+
+  #Add Min/Max labels if we aren't plotting min and max
+  line <- 0.33
+  for(ml in na.omit(names(minMaxLabels))){
+    #Extract Timezone
+    tzf <- format(as.POSIXct(minMaxLabels[[ml]][['time']]), "%z")
+    #Insert ":" before 2nd to last character
+    tzf <- sub("([[:digit:]]{2,2})$", ":\\1", tzf)
+    formatted_label <- paste0(minMaxLabels[[ml]][['legend.name']], statTimeSeries[['units']], 
+                              format(as.POSIXct(minMaxLabels[[ml]][['time']]), " %b %d, %Y %H:%M:%S"), " (UTC ", tzf, ")")
```

Comment:
could we make all of this formatting label stuff into a function? Not really sure what file it would go in. Then we would just have:

```r
line <- 0.33
for(ml in na.omit(names(minMaxLabels))){
    formatted_label <- formatMinMaxLabel(ml, statTimeSeries[['units']])
    plot_object <- ....
    line <- line + 1
}
```